compiler/quilt/tools/command.py

```diff
@@ -1507,3 +1511,208 @@ def load(pkginfo):
+            * Ensure source exists
+            * Ensure destination doesn't exist
+            * Remove absolute anchor in dest, if any
+            * Prefix destination with target dir
+            * Locate and record any zero-byte files in source (see comments)
+
+        :returns: (<source Path / dest Path pairs list>, <set of zero-byte files>)
+        """
+        # We return list instead of yielding, so that all prep logic is done before write is attempted.
+        final_export_map = []
+        zero_byte_files = set()
+
+        for src, dest in exports:
+            # general cleanup
+            src = pathlib.Path(src)
+            dest = pathlib.PureWindowsPath(dest)
+            dest = outpath.joinpath(*dest.parts[1:]) if dest.anchor else outpath / dest
+
+            # existence checks
+            # this src check replaces previous existence assertion, doubling as zero-byte file check
+            # Adam was running into an issue where shutil wasn't copying zero-byte files correctly (see below)
+            if src.stat().st_size == 0:
+                zero_byte_files.add(src)
+            if dest.parent != outpath and dest.parent.exists() and not force:
+                raise CommandException("Invalid export path: subdir already exists: {!r}"
+                                       .format(str(dest.parent)))
+            if dest.exists() and not force:
+                raise CommandException("Invalid export path: file already exists: {!r}".format(str(dest)))
+
+            final_export_map.append((src, dest))
+        return final_export_map, zero_byte_files
+
+    def check_for_conflicts(export_list):
+        """Checks for conflicting exports in the final export map of (src, dest) pairs
+
+        Export conflicts can be introduced in various ways -- for example:
+            * export-time mapping -- user maps two files to the same name
+            * coded builds -- user creates two FileNodes with the same path
+            * re-rooting absolute paths -- user entered absolute paths, which are re-rooted to the export dir
+            * build-time duplication -- user enters the same file path twice under different nodes
+
+        This checks for these conflicts and raises an error if they have occurred.
+
+        :raises: CommandException
+        """
+        # Check for file conflicts -- data coming from more than one source to the same dest.
+        files = set(dest for src, dest in export_list)
+        conflict_counter = Counter(files)
+        file_conflicts = [dest for dest, count in conflict_counter.items() if count > 1]
+        verified_file_conflicts = []
+
+        if file_conflicts:
+            # kinda slow, but only happens if a conflict has definitely occurred.
+            for conflict in file_conflicts:
+                matches = [item for item in export_list if item[1] == conflict]
+                # if all are from the same source, it's not really a conflict.
```

Comment:
What is checking? Are you testing this to allow re-exporting into the same location?

Reply:
Testing this to ensure the package doesn't have two files that export to the same name.  For example, if the user adds the contents of a particular folder to a package, but at different times (and different states) under different node names.  The export locations are still the same, because they came from the same directory tree structure.